tests/TestHarness/Node.py

```diff
@@ -156,6 +156,12 @@ def getTransBlockNum(trans):
         if cntxt.hasKey("processed"):
             cntxt.add("processed")
             cntxt.add("action_traces")
+            # action_traces could be empty when an exception happened in the transaction
+            # cntxt.index(0) below will crash action_traces is empty
+            cur=cntxt.getCurrent()
+            assert isinstance(cur, list), f"ERROR: action_traces is not a list."
+            if len(cur) == 0:
+                return "no_block"
             cntxt.index(0)
             if not cntxt.isSectionNull("except"):
                 return "no_block"
```

Comment:
Could you instead just moved this `if` for `except` up before the `cntxt.add("action_traces")`

Reply:
Correct.

Reply:
Thanks!